File: library-net/src/main/java/com/my/library_net/NetModuleInit.java
package com.my.library_net;

import android.app.Application;
import android.content.Context;

import com.my.library_base.init.IModuleInit;
import com.my.library_base.logs.GLog;


/**
 * Created by yan
 * 基础库自身初始化操作
 */

public class NetModuleInit implements IModuleInit {
    public static Context applicationContext;

    @Override
    public boolean onInitAhead(Application application) {
        GLog.i("Net层初始化 -- onInitAhead");
        initData(application);
        return true;
    }

    @Override
    public boolean onInitLow(Application application) {
        GLog.i("Net层初始化 -- onInitLow");
        return true;
    }

    private void initData(Application application) {
        applicationContext = application.getApplicationContext();
    }

}

File: library-net/src/main/java/com/my/library_net/NetModuleInit.kt
package com.my.library_net

import android.app.Application
import android.content.Context
import com.my.library_base.init.IModuleInit
import com.my.library_base.logs.GLog

/**
 * Created by yan
 * 基础库自身初始化操作
 */
class NetModuleInit constructor() : IModuleInit {
    override fun onInit(application: Application): Boolean {
        GLog.i("Net层初始化 -- onInitAhead")
        initData(application)
        return true
    }

    override fun onDestroy(application: Application): Boolean {
        GLog.i("Net层销毁 -- onDestroy")
        return true
    }

    private fun initData(application: Application) {
        applicationContext = application.applicationContext
    }

    companion object {
        var applicationContext: Context? = null
    }
}